docs/source/en/optimization/coreml.mdx

```diff
@@ -0,0 +1,164 @@
+
+The rest of this guide provides additional details on Core ML conversion and inference using Apple's repo.
+
+
+## Selecting the Core ML Variant to Use
+
+There are several variants per model:
+
+- "Original" attention vs "split_einsum". These are two alternative implementations of the critical attention blocks. `split_einsum` was [previously introduced by Apple](https://machinelearning.apple.com/research/neural-engine-transformers), and is particularly optimized for Apple's Neural Engine (ANE) device, which is available in modern iPhones, iPads and M-series computers. `original` attention is only compatible with CPU and GPU, but it can be faster to run your model on CPU+GPU with original attention than it is to use the ANE. See [this performance benchmark](https://huggingface.co/blog/fast-mac-diffusers#performance-benchmarks) as well as some [additional measures provided by the community](https://github.com/huggingface/swift-coreml-diffusers/issues/31).
+
+- "ML Packages" vs "Compiled" models. The former is suitable for Python inference, while the `compiled` version is required for Swift code. The `compiled` models in the Hub split the large UNet model weights in several files for compatibility with iOS and iPadOS devices. This corresponds to the [`--chunk-unet` conversion option](https://github.com/apple/ml-stable-diffusion#-converting-models-to-core-ml). If you want to support native apps, then you need to select one of the "compiled" variants.
+
+The 4 official Stable Diffusion models include these 4 variants, but the community ones may vary. This is the structure followed by the 4 official models:
+
+```
+coreml-stable-diffusion-v1-4
+├── README.md
+├── original
+│   ├── compiled
+│   └── packages
+└── split_einsum
+    ├── compiled
+    └── packages
+```
+
+You can download and use the variant you need as shown below.
+
+## Core ML Inference in Python
+
+### Prerequisites
+
+```bash
+pip install huggingface_hub
+pip install git+https://github.com/apple/ml-stable-diffusion
+```
+
+### Download the Model Checkpoints
+
+To run inference in Python, you have to use one of the versions stored in the `packages` folders, because the compiled ones are only compatible with Swift. You may choose whether you want to use the `original` or `split_einsum` attention styles.
+
+This is how you'd download the `original` attention variant from the Hub:
+
+```Python
+from huggingface_hub import snapshot_download
+from pathlib import Path
+
+repo_id = "apple/coreml-stable-diffusion-v1-4"
+variant = "original/packages"
+
+model_path = Path("./models") / (repo_id.split("/")[-1] + "_" + variant.replace("/", "_"))
+snapshot_download(repo_id, allow_patterns=f"{variant}/*", local_dir=model_path, local_dir_use_symlinks=False)
+print(f"Model downloaded at {model_path}")
+```
+
+The code above will place the downloaded model snapshot inside a directory called `models`.
+
+### Inference
```

Comment:
Since there are identical section headers, clicking on the Inference in the Swift section defaults to the Python Inference section. You can add these tags to differentiate them:

```suggestion
### Inference[[python-inference]]
```